### DISCUSSION TITLE
v4.4.X Feedback

### DISCUSSION THREAD

**Discussion by metaljay** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#753] [General]
Just flashed the latest update, had to remove the board from the pump as when I flashed OTA it lost the wifi settings. When I reflashed from scratch I noticed you've changed the landing page for the AP. Anyway togo back to layzspa.local for initial setup rather than an IP? Is this because you've ditched the wifi manager? 

the '?' for Soft access point wasn't visible on mobile devices, only from a proper web client. 
Should Soft Access point be enabled by default by the way?
Occasionally when I refresh home I also get a button called 'Take Control', ive not seen this documented anywhere, what is it?
![snip2](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/f522a656-869b-44d1-aaa6-d549d751f61b)

The 'brightness' toggle also seems off when I update the pump config
![snip3](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/db4e3533-678d-41e7-ba9c-b7786f89fd8c)

~~MQTT data also become a little weird in home assistant, ive wiped everything including cached but still showing invalid date. It connects ok but I can't control.~~ - Ignore, a few reboots later and all working again
<img width="489" alt="Screenshot 2024-07-09 at 12 19 05 PM" src="https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/d5290f94-2d29-4363-a343-d90189bbd11c">
<img width="460" alt="Screenshot 2024-07-09 at 12 19 19 PM" src="https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/10380582/09f52046-44c9-4176-b783-7d67219da612">

as an aside, the boot up times are incredible, great job!

**Comment by visualapproach** [comment]
> When I reflashed from scratch I noticed you've changed the landing page for the AP. Anyway togo back to layzspa.local for initial setup rather than an IP? Is this because you've ditched the wifi manager?

The .local has not changed. When ESP is not connected to an AP it starts its own. WiFi manager did some tricks to spawn a page automatically upon connecting to it. Now we need to go there explicitly. WM was redundant and took too much memory.

> the boot up times are incredible, great job!


Thanks!
/VA

**Reply by metaljay** [reply]
What I mean is, when flashing the firmware for the first time, previously I would connect to the AP network, and then go to Layzspa.local to configure. That no longer works, I have to use the IP address. 
Are you able to change it?

What about take control feature?

**Reply by visualapproach** [reply]
Oh, I think you could go to any address and WM redirected you to its config page.
I don't know if I can do it. It's not high on the priority list.
'Take control' is mentioned briefly in the manual but it's only for 4-wire models. Basically you have to choose whether the display or ESP will control the pump.

**Comment by visualapproach** [comment]
Brightness slider is back in shape again :-) (dev.branch)

**Reply by metaljay** [reply]
> Brightness slider is back in shape again :-) (dev.branch)

Tested 4.4.2, all good so far.

As an aside, is there an easier way to update the firmware and maintain wifi / spa config? 
each time I flash via 'upload_protocol = espota' and flash the filesystem it wipes my settings, should I be doing something different?

**Reply by visualapproach** [reply]
You could upload the data files one by one from the UI. It will preserve all save-files like "wifi.json". I haven't found a good way to upload several files at once. There is also a possibility to store settings in "EEPROM" which I believe will survive a filesystem flashing. But it has other caveats.

**Reply by metaljay** [reply]
I haven’t looked at the coding, but is it not possible to store the WiFi credentials in another field/ file so that when you flash via OTA the link to the file which stores credentials always leaves said file alone?

home assistant does something similar with updates

(clearly not a programmer)

**Reply by visualapproach** [reply]
No it is not possible. When uploading an image you wipe the whole section of Flash memory that holds the filesystem.
But it is possible to upload/overwrite files to the system. THIS is what HA does. But the ESP has 30 kB free to manage that whilst HA runs on a computer with gigabytes free. The ESP cannot handle several uploads at once.

**Reply by visualapproach** [reply]
Well, that said, I found a super easy way to solve the connection issue after uploading fs image 😄 
Turned out that the esp saves last known credentials in "EEPROM" (another section of the flash in this case)
So I just had to use that. You'd have to set all other thing though, unless you first download the files (settings.json, mqtt.json hwcfg.json, webconfig.json) and reload your saved commandqueue if you have one. And then copy those files to the data directory before making the file system image.

So I think you'll like the dev branch now!
